.github/workflows/publish.yml

```diff
@@ -1,25 +1,48 @@
 name: publish
+
 on:
+  workflow_dispatch: # allow manual runs
   push:
     branches:
       - main
-  workflow_dispatch:
+
 permissions:
-  contents: write
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: github-pages
+  cancel-in-progress: false # skip any intermediate builds but let finish
+
 jobs:
-  deploy:
+  build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: 3.11
+      - id: pages
+        uses: actions/configure-pages@v4
       - run: echo "cache_id=$(date --utc '+%V')" >> $GITHUB_ENV
-      - uses: actions/cache@v3
+      - uses: actions/cache@v4
         with:
           key: mkdocs-material-${{ env.cache_id }}
           path: .cache
```

Comment:
FYI the only thing I'm not exactly satisfied with is the cache. I haven't touched that bit, but TL;DR is doesn't work at all.

I see the caching key here is week number, and the attempt to store `.cache` is common for various package manager resolves. Unfortunately here the cache action stores `/home/runner/work/kitsune/kitsune/.cache` (empty), whereas the built python wheels that could be worth caching are stored in `/home/runner/.cache/pip/wheels` out of reach. These _could_ shave a few seconds off the clock. But with the frequency this is gonna be run it doesn't really make that much difference TBH…